src/cascadia/TerminalCore/TerminalSelection.cpp

```diff
@@ -19,77 +19,116 @@ std::vector<SMALL_RECT> Terminal::_GetSelectionRects() const
+    // Here, we want the "left"most coordinate to be the one "higher" on the screen. The other gets the dubious honor of
+    // being the "lower."
+    return bufferSize.CompareInBounds(selectionAnchorWithOffset, endSelectionPositionWithOffset) <= 0 ?
+               std::make_tuple(selectionAnchorWithOffset, endSelectionPositionWithOffset) :
+               std::make_tuple(endSelectionPositionWithOffset, selectionAnchorWithOffset);
+}
+
+// Method Description:
+// - convert selection anchors to proper coordinates for rendering
+// Arguments:
+// - row: the buffer y-value under observation
+// - higherCoord: the higher coordinate (or leftmost)
+// - lowerCoord: the lower coordinate (or rightmost)
+// Return Value:
+// - tuple.first: the coordinate the higher coordinate (or leftmost)
+// - tuple.second: the lower coordinate (or rightmost)
+SMALL_RECT Terminal::_GetSelectionRow(const SHORT row, const COORD higherCoord, const COORD lowerCoord) const
+{
+    SMALL_RECT selectionRow;
+
+    selectionRow.Top = row;
+    selectionRow.Bottom = row;
+
+    if (_boxSelection || higherCoord.Y == lowerCoord.Y)
     {
-        SMALL_RECT selectionRow;
+        selectionRow.Left = std::min(higherCoord.X, lowerCoord.X);
+        selectionRow.Right = std::max(higherCoord.X, lowerCoord.X);
+    }
+    else
+    {
+        selectionRow.Left = (row == higherCoord.Y) ? higherCoord.X : 0;
+        selectionRow.Right = (row == lowerCoord.Y) ? lowerCoord.X : _buffer->GetSize().RightInclusive();
+    }
 
-        selectionRow.Top = row;
-        selectionRow.Bottom = row;
+    return selectionRow;
+}
 
-        if (_boxSelection || higherCoord.Y == lowerCoord.Y)
-        {
-            selectionRow.Left = std::min(higherCoord.X, lowerCoord.X);
-            selectionRow.Right = std::max(higherCoord.X, lowerCoord.X);
-        }
-        else
-        {
-            selectionRow.Left = (row == higherCoord.Y) ? higherCoord.X : 0;
-            selectionRow.Right = (row == lowerCoord.Y) ? lowerCoord.X : bufferSize.RightInclusive();
-        }
+// Method Description:
+// - Expand the selection row according to selection mode and wide glyphs
+// - this is particularly useful for box selections (ALT + selection)
+// Arguments:
+// - selectionRow: the selection row to be expanded
+// Return Value:
+// - modifies selectionRow's Left and Right values to expand properly
+void Terminal::_ExpandSelectionRow(SMALL_RECT& selectionRow) const
+{
+    const auto row = selectionRow.Top;
 
-        // expand selection for Double/Triple Click
-        if (_multiClickSelectionMode == SelectionExpansionMode::Word)
+    // expand selection for Double/Triple Click
+    if (_multiClickSelectionMode == SelectionExpansionMode::Word)
+    {
+        const auto cellChar = _buffer->GetCellDataAt({ _selectionAnchor.X, row })->Chars();
+        if (_isSingleCellSelection() && _isWordDelimiter(cellChar))

```

Comment:
this may be a trivial time and place to fix #2206 too

Reply:
Done. Makes the code look nicer too.